cloud-nio/cloud-nio-impl-drs/src/main/scala/cloud/nio/impl/drs/DrsPathResolver.scala

```diff
@@ -133,4 +140,24 @@ object MarthaResponseSupport {
       hashes = hashesMap
     )
   }
+
+  def errorMessageFromResponse(drsPathForDebugging: String, marthaResponseEntityOption: Option[String], responseStatusLine: StatusLine, marthaUri: String): String = {
+    val baseMessage = s"Could not access object \'$drsPathForDebugging\'. Status: ${responseStatusLine.getStatusCode}, reason: \'${responseStatusLine.getReasonPhrase}\', Martha location: \'$marthaUri\', message: "
+
+    marthaResponseEntityOption match {
+      case Some(entity) =>
+        val maybeErrorResponse: Either[Error, MarthaFailureResponse] = decode[MarthaFailureResponse](entity)
+        maybeErrorResponse match {
+          case Left(_) =>
+            // Not parsable as a `MarthaFailureResponse`
+            baseMessage + s"\'$entity\'"
+          case Right(decoded) =>
+            // Is a `MarthaFailureResponse`
+            baseMessage + s"\'${decoded.response.text}\'"
+        }
+      case None =>
+        // No entity in HTTP response
+        baseMessage + "(empty response)"
+    }
```

Comment:
This is a pretty simplistic and verbose factoring, we could probably make use of `IO` somehow to make it cooler. Open to suggestions.

Reply:
I'm not gonna worry about it though, since I've received two "good enough"'s